src/CombineUnixArchives.cmake

```diff
@@ -68,16 +68,24 @@ function(COMBINE_UNIX_ARCHIVES
 
     # Extract objects from current source lib
     execute_process(COMMAND ${CMAKE_AR} -x ${lib}
-                    WORKING_DIRECTORY ${working_dir})
+                    WORKING_DIRECTORY ${working_dir}
+                    RESULT_VARIABLE exe_result)
+    if(NOT "${exe_result}" STREQUAL "0")
+      message(FATAL_ERROR "COMBINE_UNIX_ARCHIVES: obj extraction process failed exe_result='${exe_result}'")
+    endif()
 
     # Accumulate objects
-    file(GLOB_RECURSE objects "${working_dir}/*")
+    file(GLOB_RECURSE objects "${working_dir}/*.o")
```

Comment:
`ar -x` also dumps the archive index file (e.g., `__.SYMDEF SORTED`), which was getting picked up by `file(GLOB_RECURSE objects "${working_dir}/*")`, and subsequently added to the new archive via `ar rcs`. This resulted in invalid archive in systems with the newer OS X command line toolchain (e.g., on Sierra or when xcode 8 was installed on older OS X), but not with the older toolchain.